src/librustc_resolve/error_codes.rs

```diff
@@ -1661,6 +1661,30 @@ match eco {
 ```
 "##,
 
+E0578: r##"
+A module cannot be found and therefore, the visibility cannot be determined.
```

Comment:
The comma here is odd.  Either 'therefore' is an optional part of the sentence in which case it needs a comma before as well as after, or it's not optional and the comma is unnecessary.

e.g.

> A module cannot be found and, therefore, the visibility cannot be determined.

vs.

> A module cannot be found and therefore the visibility cannot be determined.